analyzers/tests/SonarAnalyzer.UnitTest/Helpers/FilesToAnalyzeProviderTest.cs

```diff
@@ -0,0 +1,112 @@
+        private const string InvalidFilesToAnalyzePath = @"ResourceTests\FilesToAnalyze\InvalidValuesFileToAnalyze.txt";
+
+        [TestMethod]
+        public void FileNameWithMixedCapitalizationAndMixedSlashes_FindFilesWithFileName_ReturnsAllWebConfigFiles()
+        {
+            var sut = new FilesToAnalyzeProvider(FilesToAnalyzePath);
+
+            var results = sut.FindFiles("Web.config");
+            results.Should().HaveCount(2);
+            results.Should().Contain(MixedSlashesWebConfigPath1);
+            results.Should().Contain(MixedSlashesWebConfigPath2);
+        }
+
+        [TestMethod]
+        public void FileNameWithMixedCapitalizationAndMixedSlashes_FindFilesWithRegex_ReturnsAllWebConfigFiles()
+        {
+            var fileNamePattern = new Regex("web\\.config$", RegexOptions.IgnoreCase);
+
+            var sut = new FilesToAnalyzeProvider(FilesToAnalyzePath);
+
+            var results = sut.FindFiles(fileNamePattern);
+            results.Should().HaveCount(2);
+            results.Should().Contain(MixedSlashesWebConfigPath1);
+            results.Should().Contain(MixedSlashesWebConfigPath2);
+        }
+
+        [TestMethod]
+        public void FileWithInvalidvalues_FindFilesWithFileName_ReturnsValidValue()
+        {
+            var sut = new FilesToAnalyzeProvider(InvalidFilesToAnalyzePath);
+
+            var results = sut.FindFiles("Web.config");
+            results.Should().HaveCount(1);
+            results.Should().Contain(MixedSlashesWebConfigPath2);
+        }
+
+        [TestMethod]
+        public void FileWithInvalidvalues_FindFilesWithRegex_ReturnsValidValue()
+        {
+            var fileNamePattern = new Regex("web\\.config$", RegexOptions.IgnoreCase);
+
+            var sut = new FilesToAnalyzeProvider(InvalidFilesToAnalyzePath);
+
+            var results = sut.FindFiles(fileNamePattern);
+            results.Should().HaveCount(1);
+            results.Should().Contain(MixedSlashesWebConfigPath2);
+        }
+
+        [TestMethod]
+        public void NonExistentFile_FindFiles_ReturnsEmptyEnumerable()
+        {
+            var sut = new FilesToAnalyzeProvider(@"ResourceTests\FilesToAnalyze\NonExistingFile.txt");
+
+            var results = sut.FindFiles("Web.config");
+            results.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void EmptyFile_FindFiles_ReturnsEmptyEnumerable()
+        {
+            var sut = new FilesToAnalyzeProvider(@"ResourceTests\FilesToAnalyze\EmptyFilesToAnalyze.txt");
+
+            var results = sut.FindFiles("Web.config");
+            results.Should().BeEmpty();
+        }
+
+        [DataTestMethod]
+        [DataRow("")]
+        [DataRow(null)]
+        [DataRow("invalidPath")]
+        public void InvalidPath_FindFiles_ReturnsEmptyEnumerable(string filePath)
+        {
+            var sut = new FilesToAnalyzeProvider(filePath);
+
+            var results = sut.FindFiles("Web.config");
```

Comment:
Same as above for empty
```suggestion
            var results = sut.FindFiles(new Regex(".*"));
```